### PR TITLE
Remove datalink template suggestions for accessing specific fields when there are multiple dataframes.

### LINKED ISSUE #31200: Data links: Auto suggest for fields shows display name instead of the real field name
Hello,

I want to use the value from the _specific_ series to compute data link in the graph panel. It looks like the editor provides a suggestion for that, but the engine doesn't resolve the template into real value.

Steps to reproduce:
1. Open [https://play.grafana.org/d/000000012/grafana-play-home?orgId=1](https://play.grafana.org/d/000000012/grafana-play-home?orgId=1)
2. Edit "server requests" panel
3. Click "Field" tab
4. Click "Add Link" button
5. Set something to the title (e.g. "test link")
6. Type $ at URL field and wait for the popup (code completion)
7. Select "Fields web_server_01" item. URL should be `${__data.fields.web_server_01}`
8. Click OK
9. Click on the graph and ensure that the expression in the test link is not resolved.

image

On the other hand `${__data.fields.Time}` expression is resolved correctly:

image

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What this PR does / why we need it**:
When there are multiple DataFrames the fields suggestions for the datalinks does not work. This removes those suggestions in that case. It also makes sure we match the displayNames of timeSeries data that have the fields Time and Value.

**Which issue(s) this PR fixes**:
Fixes #31200 (partially)

### CODE REVIEW

File: public/app/features/panel/panellinks/link_srv.ts [hunk 1]
```diff
@@ -134,30 +134,36 @@ export const getDataFrameVars = (dataFrames: DataFrame[]) => {
...
+  if (dataFrames.length !== 1) {
+    // It's not possible to access fields of other dataframes. So if there are multiple dataframes we need to skip these suggestions.
+    // Also return early if there are no dataFrames.
+    return [];
+  }
 
-      if (keys[displayName]) {
-        continue;
-      }
+  const frame = dataFrames[0];
```
Comment (oscarkilhed) [inline public/app/features/panel/panellinks/link_srv.ts:143]: Just removed the outer loop here as there was no need to loop over an array we know to contain only one element

**Review by hugohaggmark — APPROVED** [review]
Looks good and works as expected. 

Although when I've 2 data frames and add a data link everything looks good, if I then remove a data frame (query) the data link editor still doesn't show the expected suggestions. But if I do a full page reload, then the suggestions are correct. Not sure this an issue with this PR but probably something that has always been like this.
bug1

File: packages/grafana-data/src/field/getFieldDisplayValuesProxy.ts [hunk 2]
```diff
@@ -28,9 +28,13 @@ export function getFieldDisplayValuesProxy(
...
       }
       if (!field) {
-        // 3. Match the title
+        // 3. Match the config displayName
         field = frame.fields.find((f) => key === f.config.displayName);
       }
+      if (!field) {
+        // 4. Match the state displayName
+        field = frame.fields.find((f) => key === f.state?.displayName);
+      }
```
[the remark this answers is not in the document]
Comment (oscarkilhed) [reply packages/grafana-data/src/field/getFieldDisplayValuesProxy.ts:37]: > Although when I've 2 data frames and add a data link everything looks good, if I then remove a data frame (query) the data link editor still doesn't show the expected suggestions. But if I do a full page reload, then the suggestions are correct. 

Yeah I noticed this too. A full page reload is not required, it's enough swap to the "panel" tab and back. Not sure what the best way to solve this is. Either we move the suggestions to populate/re-populate when the user clicks the add/edit link, or we re-populate when the query changes.

[the remark this answers is not in the document]
Comment (hugohaggmark) [reply packages/grafana-data/src/field/getFieldDisplayValuesProxy.ts:37]: As I mentioned earlier, that might be something for a follow up PR instead?

**Pull request merged**